src/output.rs

```diff
@@ -174,11 +174,47 @@ impl fmt::Debug for FnPredicate {
     }
 }
 
+#[derive(Debug, Clone)]
+struct RegexPredicate {
+    regex: regex::Regex,
+    times: u32,
+}
+
+impl RegexPredicate {
+    fn verify(&self, got: &[u8]) -> Result<()> {
+        let conversion = String::from_utf8_lossy(got);
+        let got = conversion.as_ref();
+        if self.times == 0 {
```

Comment:
What are your thoughts on using a sentinel value (`0`) compared to using `Option`?

Reply:
This is a good suggestion, It's just some bad habit's take a while to leave :)

Reply:
I ask not to point out bad habits because sometimes the line is blurry.  In another project, I have one behavior when a `Vec` is empty and another when it has items.  I'm not using an `Option` for it because it doesn't feel like it'd jive quite right with the API.